pkg/querier/batch/batch.go

```diff
@@ -38,7 +38,7 @@ func (c GenericChunk) Iterator(reuse chunk.Iterator) chunk.Iterator {
 
 // iterator iterates over batches.
 type iterator interface {
-	// Seek to the batch with sample at (or after) time t.
+	// Seek advances the iterator forward to the sample at or after the given timestamp.
```

Comment:
Doesn't `Seek` advance to a batch, considering the iterator operates at the batch granularity? E.g., `Next` moves to the next batch, not sample. Wondering if the previous wording isn't more descriptive in this respect.

Reply:
`Seek` does both: it advances to the batch with a sample at or after the given timestamp, and advances within that batch to the sample at or after the given timestamp by updating `batch.Index`. 

I've tried to make this clearer in 0ee758f - how's this now?